ironic/workflow.go

```diff
@@ -100,25 +102,23 @@ func (workflow *provisionStateWorkflow) toManageable() (done bool, err error) {
 	default:
 		return true, fmt.Errorf("cannot go from state '%s' to state 'manageable'", state)
 	}
-
-	return false, nil
 }
 
 // Clean a node
-func (workflow *provisionStateWorkflow) toClean() (done bool, err error) {
+func (workflow *provisionStateWorkflow) toClean() (bool, error) {
 	// Node must be manageable first
-	workflow.reloadNode()
+	_ = workflow.reloadNode()
```

Comment:
reloadNode returns an error, what do you think about checking them? Same for changeProvisionState.  Sorry this code is some of the first go I ever wrote.   It's probably pretty sketchy.

Reply:
sure thing